src/repr/src/row.rs

```diff
@@ -501,6 +640,13 @@ impl Ord for RowRef {
     }
 }
 
+// Note that the Hash implementation must strictly follow `Row`'s hash implementation.
+impl Hash for RowRef {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.data().hash(state)
```

Comment:
The `Row` type defers to the `Hash` implementation of `CompactBytes` which indeed hashes the slice version of itself but this should be treated as an implementation detail. It can change at any moment without being a breaking change. If we have a strict requirement that `Row::hash` and `RowRef::hash` behave the same I suggest moving the `#[derive(Hash)]` onto `RowRef` and adding this `Hash` impl to `Row`:
```rust
impl Hash for Row {
    fn hash<H: Hasher>(&self, state: &mut H) {
        self.as_row_ref().hash(state)
    }
}
```

Reply:
That makes a lot of sense, changed the PR! Thanks.